src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -140,6 +146,11 @@ public void Complete()
             }
         }
 
+        public void PromiseNotToEnqueue()
```

Comment:
With `Complete` and `Cancel`, `PromiseNotToEnqueue` is unnecessary since `_disallowEnqueue` would be set by `Complete`.